test/ctmrg/pepotrace.jl

```diff
@@ -0,0 +1,130 @@
+using Test
+using Random
+using LinearAlgebra
+using TensorKit
+using KrylovKit
+using OptimKit
+using Zygote
+using PEPSKit
+import PEPSKit: unitcell
+
+pspace = ℂ^2
+vspace = ℂ^2
+χenv = 24
+T = ComplexF64
+
+Random.seed!(1564654824)
+
+# Construct random PEPO tensors
+O = randn(T, pspace ⊗ pspace', vspace ⊗ vspace ⊗ vspace' ⊗ vspace')
+M = randn(T, pspace ⊗ pspace', vspace ⊗ vspace ⊗ vspace' ⊗ vspace')
+
+# Fuse a layer consisting of O-O and MO together
+fuser = isometry(T, vspace ⊗ vspace, fuse(vspace, vspace))
+fuser_conj = isometry(T, vspace' ⊗ vspace', fuse(vspace, vspace)')
+@tensor O2[-1 -2; -3 -4 -5 -6] :=
+    O[-1 1; 2 4 6 8] *
+    O[1 -2; 3 5 7 9] *
+    fuser[2 3; -3] *
+    fuser[4 5; -4] *
+    fuser_conj[6 7; -5] *
+    fuser_conj[8 9; -6]
+@tensor MO[-1 -2; -3 -4 -5 -6] :=
+    M[-1 1; 2 4 6 8] *
+    O[1 -2; 3 5 7 9] *
+    fuser[2 3; -3] *
+    fuser[4 5; -4] *
+    fuser_conj[6 7; -5] *
+    fuser_conj[8 9; -6]
+
+# Create `InfiniteSquareNetwork`s of of both options
+network = InfiniteSquareNetwork(InfinitePEPO(fill(O, 1, 1, 2)));
+network_fused = InfiniteSquareNetwork(InfinitePEPO(O2));
+
+# cover all different flavors
+ctm_styles = [:sequential, :simultaneous]
+projector_algs = [:halfinfinite, :fullinfinite]
+
+@testset "PEPO layers CTMRG contraction using $alg with $projector_alg" for (
+    alg, projector_alg
+) in Iterators.product(
+    ctm_styles, projector_algs
+)
+    env, = leading_boundary(
+        CTMRGEnv(network, χenv), network; alg, maxiter=150, projector_alg
+    )
+    env_fused, = leading_boundary(
+        CTMRGEnv(network_fused, χenv), network_fused; alg, maxiter=150, projector_alg
+    )
+
+    m = PEPSKit.contract_local_tensor((1, 1, 1), M, network, env)
+    m_fused = PEPSKit.contract_local_tensor((1, 1, 1), MO, network_fused, env_fused)
+
+    nrm = PEPSKit._contract_site((1, 1), network, env)
+    nrm_fused = PEPSKit._contract_site((1, 1), network_fused, env_fused)
+
+    @test (m / nrm) ≈ (m_fused / nrm_fused) atol = 1e-9
+end
+
+ψ = randn(T, pspace, vspace ⊗ vspace ⊗ vspace' ⊗ vspace')
+@tensor Oψ[-1; -3 -4 -5 -6] :=
+    O[-1 1; 2 4 6 8] *
+    ψ[1; 3 5 7 9] *
+    fuser[2 3; -3] *
+    fuser[4 5; -4] *
+    fuser_conj[6 7; -5] *
+    fuser_conj[8 9; -6]
+@tensor Mψ[-1; -3 -4 -5 -6] :=
+    M[-1 1; 2 4 6 8] *
+    ψ[1; 3 5 7 9] *
+    fuser[2 3; -3] *
+    fuser[4 5; -4] *
+    fuser_conj[6 7; -5] *
+    fuser_conj[8 9; -6]
+
+(Nr, Nc) = (1, 1)
+Oinf = InfinitePEPO(O; unitcell=(Nr, Nc, 1))
+O_stack = fill(O, Nr, Nc, 2)
+O_stack[:, :, 2] .= unitcell(dagger(Oinf))
```

Comment:
```suggestion
O_stack[:, :, 2] .= unitcell(PEPSKit.dagger(Oinf))
```
Or add `dagger` as an import above.